src/config.c

```diff
@@ -3068,7 +3068,8 @@ standardConfig static_configs[] = {
     createBoolConfig("aof-disable-auto-gc", NULL, MODIFIABLE_CONFIG | HIDDEN_CONFIG, server.aof_disable_auto_gc, 0, NULL, updateAofAutoGCEnabled),
     createBoolConfig("replica-ignore-disk-write-errors", NULL, MODIFIABLE_CONFIG, server.repl_ignore_disk_write_error, 0, NULL, NULL),
     createBoolConfig("extended-redis-compatibility", NULL, MODIFIABLE_CONFIG, server.extended_redis_compat, 0, NULL, updateExtendedRedisCompat),
-    createBoolConfig("enable-debug-assert", NULL, IMMUTABLE_CONFIG | HIDDEN_CONFIG, server.enable_debug_assert, 0, NULL, NULL),
+    createBoolConfig("debug-assert-enabled", NULL, IMMUTABLE_CONFIG | HIDDEN_CONFIG, server.debug_assert_enabled, 0, NULL, NULL),
+    createBoolConfig("io_uring-enabled", NULL, DEBUG_CONFIG | MODIFIABLE_CONFIG, server.io_uring_enabled, 0, NULL, NULL),
```

Comment:
```suggestion
    createBoolConfig("io-uring-enabled", NULL, DEBUG_CONFIG | MODIFIABLE_CONFIG, server.io_uring_enabled, 0, NULL, NULL),
```
Using both dashes and underscores doesn't make much sense to me. I think basically all configs use dashes.

Reply:
The thing is: Underscore is part of the name of "io_uring". Wherever I could find any text about it online, it's written with an underscore. Of course we can change it to a dash anyway, if we want. I don't know if that's confusing to anyone.

Reply:
OK, I found a counter example to what I just said above: URLs in websites like this one: `https://stackoverflow.com/questions/61767702/what-exactly-is-io-uring`.

Thus, I think we should use underscore for the same reason, i.e. the rule "all special characters are replaced with dashes".

Reply:
> The thing is: Underscore is part of the name of "io_uring". Wherever I could find any text about it online, it's written with an underscore. Of course we can change it to a dash anyway, if we want. I don't know if that's confusing to anyone.

several months later, nobody know why the config name is io_uring-enabled,  the only thing left to people is confused:  why it is not like other configs which only contains "-" symbal.  Update to io-uring-enabled makes more sense to maintain it.

Reply:
Rename to `io-uring-enabled` as suggested.